x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/LocalStateCompositeXPackPlugin.java

```diff
@@ -153,12 +154,17 @@ protected void setLicenseState(XPackLicenseState licenseState) {
                                                NamedXContentRegistry xContentRegistry, Environment environment,
                                                NodeEnvironment nodeEnvironment, NamedWriteableRegistry namedWriteableRegistry) {
         List<Object> components = new ArrayList<>();
+        // This is a hack, but the settings we add in #additionalSettings() are not added to the environment instance
+        // (in org.elasticsearch.node.Node) which is passed in `createComponents` of each of the plugins. So the environment
+        // we get here wouldn't have the additional setting. This is a known issue, and once it is resolved, the code here
```

Comment:
I'm not sure how we track this and if we have an issue for it ( couldn't find one but I might have failed ) . There was some relevant discussion around this in https://github.com/elastic/elasticsearch/pull/49667.  Discussed it briefly with @rjernst on slack, and I didn't want to open an issue for this specific case here as it is only part of the problem.

Reply:
That doesn't sound right to me. Doesn't `environment` contain the extra settings here? I think it's supposed to.

Reply:
Yes environment should have the extra settings. My comment in slack was about the Settings passed into Plugin constructors.